modules/dnn/test/test_vkcom_layer.cpp

```diff
@@ -0,0 +1,390 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2017, Intel Corporation, all rights reserved.
+// Third party copyrights are property of their respective owners.
+
+// This tests doesn't require any external data. They just compare outputs of
+// layers using different computation backends. Input and parameters are random.
+
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+
+using namespace cv;
+using namespace cv::dnn;
+using namespace testing;
+
+static void test(Mat& input, Net& net, Backend backendId, Target targetId, bool skipCheck = false)
```

Comment:
I think it's better to add Vulkan target tests to existing ones. Please see
[test_backends.cpp](https://github.com/opencv/opencv/blob/master/modules/dnn/test/test_backends.cpp) (whole networks)
[test_layers.cpp](https://github.com/opencv/opencv/blob/master/modules/dnn/test/test_layers.cpp)
[test_halide_layers.cpp](https://github.com/opencv/opencv/blob/master/modules/dnn/test/test_halide_layers.cpp) (there are not only Halide tests actually)